### ISSUE TITLE
Market buy estimations are inverted

### ISSUE THREAD

**Issue by hdevalence** [penumbra-zone/dex-explorer#221]
![Screenshot 2024-12-16 at 2 51 08 PM](https://github.com/user-attachments/assets/3b25fd27-01b1-432c-b5eb-4d1ae71aa68f)

Since UM is above $1, "pay with" should be 1.0955 USDC

**Comment by VanishMax** [comment]
@hdevalence this seems to be correct – it's not that the price is inverted but we simulate trade and some amounts cause large price impact that makes average price slightly different from the market price.

Maybe we should add more info about the price impact in the tooltip?

<img width="326" alt="Image" src="https://github.com/user-attachments/assets/546a0939-6010-4b55-95b9-41daf3437d76" />

**Comment by hdevalence** [comment]
No, I don't think so. If 1 UM = 1.0955 USDC then if I put "1 UM" into the Buy field, I should expect to spend 1.0955 USD. The data shown in the screenshot is not internally consistent.
